packages/firestore/src/local/indexeddb_query_cache.ts

```diff
@@ -97,36 +97,75 @@ export class IndexedDbQueryCache implements QueryCache {
     );
   }
 
-  addQueryData(
+  private saveQueryData(
     transaction: PersistenceTransaction,
     queryData: QueryData
   ): PersistencePromise<void> {
-    const targetId = queryData.targetId;
-    const addedQueryPromise = targetsStore(transaction).put(
-      this.serializer.toDbTarget(queryData)
-    );
-    if (targetId > this.metadata.highestTargetId) {
-      this.metadata.highestTargetId = targetId;
-      return addedQueryPromise.next(() =>
-        globalTargetStore(transaction).put(DbTargetGlobal.key, this.metadata)
-      );
-    } else {
-      return addedQueryPromise;
+    return targetsStore(transaction).put(this.serializer.toDbTarget(queryData));
+  }
+
+  // Updates the in-memory version of the metadata. Saving is done separately.
+  // Returns true if there were any changes to the metadata.
+  private updateMetadata(queryData: QueryData): boolean {
+    let needsUpdate = false;
+    if (queryData.targetId > this.metadata.highestTargetId) {
+      this.metadata.highestTargetId = queryData.targetId;
+      needsUpdate = true;
     }
+
+    // TODO(gsoltis): add sequence number check
```

Comment:
Ahh. I guess maybe this is what motivates having this helper?  I guess I'd still advocate renaming (eventually to updateTargetIdAndSequenceNumberGlobals or whatever)

Reply:
It was originally `updateMetadataForQueryData`, but it was requested that I change it. I'd rather not spell out exactly what is getting updated in the method name, since it's conceivable that we might someday track other metadata as well. `updateMetadataFromQueryData`?

Reply:
I went with `updateMetadataFromQueryData` to make it clear the updates are coming from the passed in `QueryData`. I'm open to other suggestions though.

Reply:
Thanks, I think this is much clearer!